nodebuilder/share/module.go

```diff
@@ -50,70 +122,23 @@ func ConstructModule(tp node.Type, cfg *Config, options ...fx.Option) fx.Option
 			// cacheAvailability's lifecycle continues to use a fx hook,
 			// since the LC requires a cacheAvailability but the constructor returns a share.Availability
 			fx.Provide(cacheAvailability[*light.ShareAvailability]),
+			fx.Invoke(ensurePeersLifecycle),
 		)
-	case node.Bridge, node.Full:
+	case node.Bridge:
 		return fx.Module(
 			"share",
 			baseComponents,
-			fx.Invoke(func(edsSrv *shrexeds.Server, ndSrc *shrexnd.Server) {}),
-			fx.Provide(fx.Annotate(
-				func(host host.Host, store *eds.Store, network modp2p.Network) (*shrexeds.Server, error) {
-					return shrexeds.NewServer(host, store, shrexeds.WithProtocolSuffix(string(network)))
-				},
-				fx.OnStart(func(ctx context.Context, server *shrexeds.Server) error {
-					return server.Start(ctx)
-				}),
-				fx.OnStop(func(ctx context.Context, server *shrexeds.Server) error {
-					return server.Stop(ctx)
-				}),
-			)),
-			fx.Provide(fx.Annotate(
-				func(
-					host host.Host,
-					store *eds.Store,
-					getter *getters.IPLDGetter,
-					network modp2p.Network,
-				) (*shrexnd.Server, error) {
-					return shrexnd.NewServer(host, store, getter, shrexnd.WithProtocolSuffix(string(network)))
-				},
-				fx.OnStart(func(ctx context.Context, server *shrexnd.Server) error {
-					return server.Start(ctx)
-				}),
-				fx.OnStop(func(ctx context.Context, server *shrexnd.Server) error {
-					return server.Stop(ctx)
-				}),
-			)),
-			fx.Provide(fx.Annotate(
-				func(path node.StorePath, ds datastore.Batching) (*eds.Store, error) {
-					return eds.NewStore(string(path), ds)
-				},
-				fx.OnStart(func(ctx context.Context, store *eds.Store) error {
-					err := store.Start(ctx)
-					if err != nil {
-						return err
-					}
-
-					return ensureEmptyCARExists(ctx, store)
-				}),
-				fx.OnStop(func(ctx context.Context, store *eds.Store) error {
-					return store.Stop(ctx)
-				}),
-			)),
-			fx.Provide(fx.Annotate(
-				full.NewShareAvailability,
-				fx.OnStart(func(ctx context.Context, avail *full.ShareAvailability) error {
-					return avail.Start(ctx)
-				}),
-				fx.OnStop(func(ctx context.Context, avail *full.ShareAvailability) error {
-					return avail.Stop(ctx)
-				}),
-			)),
-			// cacheAvailability's lifecycle continues to use a fx hook,
-			// since the LC requires a cacheAvailability but the constructor returns a share.Availability
-			fx.Provide(cacheAvailability[*full.ShareAvailability]),
-			fx.Provide(func(shrexSub *shrexsub.PubSub) shrexsub.BroadcastFn {
-				return shrexSub.Broadcast
+			bridgeAndFullComponents,
+			fx.Provide(func(ipldGetter *getters.IPLDGetter) share.Getter {
+				return ipldGetter
 			}),
+			fx.Invoke(ensurePeersLifecycle),
```

Comment:
common? Fulls also has to call it. Everyone calls ensure, only Full and Bridge advertise

Reply:
full calls implicitly in peer manager

Reply:
ah, so scattered

Reply:
not anymore